core/os/input_event.cpp

```diff
@@ -62,10 +62,16 @@ bool InputEvent::is_action(const StringName &p_action) const {
 
 bool InputEvent::is_action_pressed(const StringName &p_action) const {
 
-	return false; // InputMap::get_singleton()->event_is_action(Ref<InputEvent>(this),p_action);
+	if (is_action(p_action)) {
+		return is_pressed();
+	}
+	return false;
 }
 bool InputEvent::is_action_released(const StringName &p_action) const {
 
+	if (is_action(p_action)) {
+		return !is_pressed();
```

Comment:
No, you're calling this on individual `InputEvents`. That would be the case if it was implemented this way for `Input.is_action_released(action)`, yes.
But since this is for events, there's just no way `is_action_released` will return true if the event is something other than specified in the action.